pkg/acme/client/middleware/logger.go

```diff
@@ -18,13 +18,18 @@ package middleware
 
 import (
 	"context"
+	"time"
 
 	"golang.org/x/crypto/acme"
 	"k8s.io/klog"
 
 	"github.com/jetstack/cert-manager/pkg/acme/client"
 )
 
+const (
+	timeout = time.Second * 30
```

Comment:
I feel like 30s is quite a long time for us to be waiting here. In a lot of cases, we'll make one request in <1s, be told to 'retry-after' 10 minutes or so, and then we'll sit around doing nothing for 29s. Was there any particular reason for 30s other than it being a not too big, not too small value?

Reply:
Yes, basically goldylocks. Want me to bump it down to 10s?